### DISCUSSION TITLE
Querying a Model without withAll or with or withAllRecursive still gives back relations

### DISCUSSION THREAD

**Discussion by pwfraley** [CodeDredd/pinia-orm#1415] [Q&A] [answered]
Hi All,

I encountered a problem with pinia which is really weird.

We have a Model: Expertise

{
      id: this.uid(null),
      appraiserId: this.uid(null),
      appraiser: this.belongsTo(User, "appraiserId"),
}

Now when I query the expertise repository like so:

this.expertiseRepo.query().whereId('ID').first()

I would expect to get the following object back:

{
  id: 'VERYLONGUID'
  appraiserId: 'ID of the Appraiser'
}

But I am getting the following object back:

{
  id: 'VERYLONGUID'
  appraiserId: 'ID of the Appraiser'
  appraiser: {
    username: 'HANS@WURST.DE'
  }
}

Why is appraiser object included even though I do not have a with or withAll or withAllRecursive in the query?

TIA

**Comment by pwfraley** [comment]
One more thing I discovered is that the query works on first call.  When I then change the ExpertiseObject and do a put to the backend, after that all related objects come back with the query eventhough I am not asking it to output the related objects

**Reply by CodeDredd** [reply]
Hey @pwfraley ,

sorry for the late answer. I think this sounds like a hydration cache problem. I first need an reproduction of it.

**Reply by michastubi** [reply]
Hmm i think i am stuck at the same problem. Or I don´t get it how this should work. I made an reproduction in the playground. Is this the expected behaviour?

 
https://pinia-orm-play.codedredd.de/#N4IgDghgxg1hDmBTAziAXAbVAOwgW0XRADcBXRAWgBNE8BLEAGhGQHtSAnKQtEU7MDHhMQNZFA50wAFzqtsRAGrkABABFadFQAoAFhAA2AMxV1sKgGIdE2KLoCUp5CogqaxRAdZgz8FaVkDOmkATxcDLwB3ZxD2FWlWFUjJaUQVAFVsOg8OZEMVZTSAGToAIw4ISRQVI1YOAtUAJhUAMhUAZhFOAyJdaWkwZDQAemH+QXgAOihWPGGyShp6YaDS4bMaAA9JvAArVABfRhx8HhAAAQXSZERh5H1rKhE2Tm4icaERMQkpWXkiADKD0QVAaiHSN38gWCdBQky6HB6vD6AyGow+Uxmc0u5Gut3ulRB5wArJMAAyTACM62wW0mKDwk32ICOJwIRBxiDxwxm1me7C4ZwxXxQPxkcgUvAAwqwIogoH9zKwTCgbthZPlCioZXgwKxkDD5BlobIUAikSAUYMRmMBEJprNhpzubzECTyVSaXSGUzDscQLh2bxOcN3AlZcgKBAfPzXkK7cJmN9JOL-rwAJLqxAVBVJYK6eK6NJasOsCM1DizQtpSBIc29frW9EJh3YhahxDEcMGSPRujnABsk0antWwwZXsQ22ZrIDpyIPiyEFjgveCZF4hTiqIAAUzHRl8xug3UTaMa3houD+cR5TJgPQ3RkNJL-uIPTkHgKOVWNFs76WX9QMznOK8IAoOo5mwSDDDoAAvDgVzeXhhSTUUtwlXc3xUAB5AAlABZFRoI4PBYIQ+tkUbNFbQmC9QLfCDSOGEiyKCBDzjvAB2clH2fSdtj2P02TOXBoLoKgWPkCgbigThCGYF5VxQ9c0M3X5MN4AA5CBxKeI9ERPJtaPtLEWN01gJPOAAWD0ySk7AZPleSBIA2dgIXRjIKQ+MJg3MVt14Pcl1wwjKMtaizxbMywKYvBOMmHj7KoJ8Xw2KcdhnABdZgjDoAwzTQUAAEEwDASYFnQUBUl1AwIFSIgAB4UuIAA+AAdcxmuyVrgGAPFcgOA5GsfNrOpUFRuravqBuQAB1fMABVWCofUhpGlqOuwDaeueAKiDoXU6mkFRgBULEwACEEVAOCsqwAcgWe7OsOvUOBOs68TwxA9Ruu68BUe7Ysg56siO96MhueojErAH7smYYIX-fZQde463Hq1wYYehGqEx31Qc6mZsGffwoe+36AF4ycQCnWG0JGOHsTrOoGunJjyDxtDx6R31m5nsCJ+RSdmlRqYuq6qG0bRHEp1qaY4dnIiLaxtHakAJPVxgVEpexJiQaQZYF4mRah+alpW-UxfO2ZLtSKWZbF+W2Z+1hJkifNtHuhJVuQe69eV7NEDVjWqC1nW9YNo3ANEfHmSKkAarAOqGt4PatyIKc3pOmgjAgUgDA+8aFaGFQMGLibgAribTCoNAdcYauJuA+v7oAKVYXRsAAaXuxvzBriaIGITGOHTPB4HSRFW6tGi6EmMAKhH3mOGmKBhnaMkyQAfkO+BKeJLi+6bmo6FyaQdIIVuO674+B8Hurn0vxBW97-vB4mn39Xr8v74-qu-4f3iMEAqrcASkDANmO+QCJpHBPpXeBNdZDSFAYDGUspoEwLgYAlQOVq7YMHgAoBEl66NHfh-FugM266UQGoTBg9h6j3HpPaegNZ42nnovRhK814by3rvCelNrLEnoTXPK59n7XxoaIiaj8L6nFbmoVgiAZHxEtqXX+MDTqIM-iAl+gMtTtCcG4eQKjyFAIITAohWjgEoP0fdb6BUICQi0qQTYJ12iqNgeY-+OjbGoPusVKgoJnykCMEYLxN0fETTwX-A4xcsqdQODHRmAEE5JxTmcdOvwDrgw+ioAiK1PB-WxnDYGpFUZ5JUMtVa-1AYIxqW7FGLNsBZ3RrnfOhdzqP2cIzFQU5Ui0mcIUmgBhtED2fPVOgUB+nqmCGEam91ZqEwmbzWQMy8qeCoMgR21iJrWGkJwcweya4kMLE+SYpAJIy2icRBR5zkAc2kJIbA8Avb+1udwyozCp4GHrtIXQFznwvLefdD5J9xFP3uQCoFzzfDvPsLcuRkiHlPJBQi25X9S4wsefoZABFdIhG0I07WSyobpioOCuJxd4nYGSUcEAjS0nVVoMneqWTFL7V4GjCGZ0RnFNuqUwG5S8ArLaRDDpBcTpQB6dUy2-T3E2G2QUopYzrGTPWbM5BCzAZYpWRNDV0zT5bJ2Y4E5ByjnjOIXXVFVypaIpPsg1BOK0XwrBQ6nBA0KX-MBY8+qzztDYALgYD1g9aWwKSSyIaQA

**Reply by CodeDredd** [reply]
Thanks for the reproduction. Going to create a bug ticket out of it.

**Reply by danh65** [reply]
I just ran across this exact issue. I had no idea this was happening until I ran into a bug in our system and traced it back to pinia orm.
Is there an update on when this will be addressed? Is there a bug ticket created? I wasn't able to find one. We need it fixed asap or we'll have to find another solution.
Thanks!

**Reply by CodeDredd** [reply, marked as answer]
@danh65 I found a solution. It was pretty hard without lsing to much performance and keeping hydration correctly. 
You can test it with the latest edge release `"pinia-orm": "npm:pinia-orm-edge@latest"`

**Reply by danh65** [reply]
Great! I'll test it out and see how it goes. Thanks so much!
